File: LAB2/python/area_rectangulo.py
def calcular_area(base_texto, altura_texto):
    try:
        base = float(base_texto)
        altura = float(altura_texto)

        if base <= 0 or altura <= 0:
            raise ValueError("Error: La base y la altura deben ser mayores que cero.")

        return base * altura

    except ValueError:
        raise ValueError("Error: Debe ingresar valores numéricos válidos.")

def main():
    base = input("Ingrese la base del rectángulo: ")
    altura = input("Ingrese la altura del rectángulo: ")

    try:
        area = calcular_area(base, altura)
        print(f"Base: {base}, Altura: {altura}, Área: {area}")
    except ValueError as e:
        print(e)

if __name__ == "__main__":
    main()

File: LAB2/ejer1/AreaRectangulo.java
import java.util.Scanner;

public class AreaRectangulo {

    public static double calcularArea(String baseTexto, String alturaTexto) {
        try {
            double base = Double.parseDouble(baseTexto);
            double altura = Double.parseDouble(alturaTexto);

            if (base <= 0 || altura <= 0) {
                throw new IllegalArgumentException("Error: La base y la altura deben ser mayores que cero.");
            }

            return base * altura;

        } catch (NumberFormatException e) {
            throw new IllegalArgumentException("Error: Debe ingresar valores numéricos válidos.");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Ingrese la base del rectángulo: ");
        String base = scanner.nextLine();

        System.out.print("Ingrese la altura del rectángulo: ");
        String altura = scanner.nextLine();

        try {
            double area = calcularArea(base, altura);
            System.out.println("Base: " + base + ", Altura: " + altura + ", Área: " + area);
        } catch (IllegalArgumentException e) {
            System.out.println(e.getMessage());
        } finally {
            scanner.close();
        }
    }
}
